src/Yoti.Auth/YotiClientEngine.cs

```diff
@@ -111,11 +112,18 @@ public async Task<AmlResult> PerformAmlCheckAsync(string appId, AsymmetricCipher
             if (string.IsNullOrEmpty(authDigest))
                 throw new InvalidOperationException("Could not sign request");
 
+            string SDKversion = typeof(YotiClientEngine).GetTypeInfo()?.Assembly?.GetName()?.Version?.ToString();
+            if (SDKversion == null)
```

Comment:
I'm not clear why this would happen.
Surely it either gets the version from the assembly successfully, or it doesn't.....?

Reply:
The `?` is the "Safe Navigation Operator", which will return null if at any stage it is unable to find the requested object, rather than throwing an exception. 

Even though I don't know why this would happen either. I thought it would be preferable to add this, so that we wouldn't be throwing an exception and failing the request in the event it fails, as I'm sure we wouldn't want to stop the request from going through if there's any potential problems retrieving this version.

Reply:
We'll be better off determining whether it happens or not, and if so why.

If there is no reason for it to be there, it should not be there.